s3utils.py

```diff
@@ -91,13 +91,17 @@ def checkAwsCreds():
 
 def checkBucket(inBucket, slog, flog, argsDump, argsList):
     # Determine what kind of input we're given. Options:
-    #   bucket name   i.e. mybucket
-    #   domain name   i.e. flaws.cloud
-    #   full S3 url   i.e. flaws.cloud.s3-us-west-2.amazonaws.com
-    #   bucket:region i.e. flaws.cloud:us-west-2
+    #   bucket name                      i.e. mybucket
+    #   domain name                      i.e. flaws.cloud
+    #   S3 url - Virtual-hosted style    i.e. flaws.cloud.s3-us-west-2.amazonaws.com
+    #   S3 url - Path style              i.e. s3-us-west-2.amazonaws.com/flaw.cloud
+    #   bucket:region                    i.e. flaws.cloud:us-west-2
     
     if ".amazonaws.com" in inBucket:    # We were given a full s3 url
-        bucket = inBucket[:inBucket.rfind(".s3")]
+        if '/' in inBucket:
```

Comment:
Add a check here to make sure the slash isn't the last character in the string. I.e. `flaws.cloud.s3.amazonaws.com/`

Reply:
Changed logic to handle similar problem!